File: leetcode/367-valid-perfect-square.cpp
class Solution
{
 public:
  bool isPerfectSquare(int num)
  {
    for (int l = 1, h = num; l <= h; /* empty */) {
      long long m = l + (h - l) / 2;
      long long sqr = m * m;
      if (sqr == num) return true;
      if (sqr < num) l = m + 1;
      else h = m - 1;
    }
    return false;
  }
};

File: leetcode/367-valid-perfect-square.py
class Solution(object):
    def isPerfectSquare(self, num):
        """
        :type num: int
        :rtype: bool
        """
        i, end = 1, num
        while i <= end:
            if i * i == num:
                return True
            i += 1
            end = int(num / i)
        return False
